src/data/tabular/adult.jl

```diff
@@ -0,0 +1,68 @@
+"""
+    load_uci_adult(n::Union{Nothing, Int}=1000)
+
+Load and preprocesses data from the UCI 'Adult' dataset
+
+# Arguments
+- `n::Union{Nothing, Int}=nothing`: The number of samples to subsample from the dataset.
+
+# Returns
+- `counterfactual_data::CounterfactualData`: A `CounterfactualData` object containing the preprocessed data.
+
+# Example
+data = load_german_credit(20) # loads and preprocesses 20 samples from the Adult dataset
+"""
+function load_uci_adult(n::Union{Nothing,Int}=1000)
+    # Throw an exception if n > 1000:
+    if !isnothing(n) && n > 1000
+        throw(ArgumentError("n must be <= 1000"))
+    end
```

Comment:
I think this only applies to german creidt, but a check of ` n <= 0` should still be here

Reply:
Alright that makes sense, will change this.